src/main/java/global/govstack/mocksris/service/PaymentHubService.java

```diff
@@ -0,0 +1,280 @@
+                    new PaymentHubOnboardingBeneficiaryDetailsDTO(
+                        beneficiary.getFunctionalId(),
+                        beneficiary.getPerson().getFinancialModality().getCode(),
+                        beneficiary.getPerson().getFinancialAddress(),
+                        beneficiary.getPerson().getIban()))
+            .toList();
+
+    return new PaymentHubOnboardingBeneficiaryDTO(requestID, paymentHubDetailsDto);
+  }
+
+  @Override
+  @Transactional
+  public void updatePaymentOnboardingStatus(
+      String data, PaymentOnboardingCallbackMode onboardingMode) throws JsonProcessingException {
+
+    var cbBody = objectMapper.readValue(data, PaymentHubBeneficiaryOnboardingCallbackDTO.class);
+
+    var failedFunctionalIds =
+        cbBody.failedCases().stream()
+            .map(PaymentHubBeneficiaryOnboardingFailedCallbackDTO::payeeIdentity)
+            .toList();
+
+    List<Beneficiary> failedBeneficiaries = new ArrayList<>();
+    var beneficiaries =
+        beneficiaryRepository.findByPaymentOnboardingRequestId(cbBody.registerRequestID());
+
+    beneficiaries.forEach(
+        beneficiary -> {
+          if (failedFunctionalIds.contains(beneficiary.getFunctionalId())) {
+            beneficiary.setPaymentOnboardingStatus(PaymentOnboardingStatus.NONE);
+            failedBeneficiaries.add(beneficiary);
+          } else {
+            beneficiary.setPaymentOnboardingStatus(PaymentOnboardingStatus.ONBOARDED);
+          }
+          beneficiaryRepository.save(beneficiary);
+        });
+
+    beneficiaryRepository.saveAll(beneficiaries);
+
+    if (onboardingMode.equals(PaymentOnboardingCallbackMode.REGISTER)
+        && !failedBeneficiaries.isEmpty()) {
+      updateBeneficiary(failedBeneficiaries);
+    }
+  }
+
+  @Override
+  @Transactional
+  public void orderPayment(List<Beneficiary> beneficiaries) {
+    var beneficiaryList =
+        beneficiaryRepository.findAllById(beneficiaries.stream().map(b -> b.getId()).toList());
+
+    validateOnboardingStatus(beneficiaryList);
+
+    var requestID = UUID.randomUUID().toString();
+    var body = constructOrderPaymentRequestBody(beneficiaryList);
+
+    HttpHeaders httpHeaders = new HttpHeaders();
+    httpHeaders.add(
+        "X-CallbackURL",
+        paymentHubProperties.callbackBaseUrl() + "/api/v1/payment/payment-callback");
+    httpHeaders.add(
+        "X-Registering-Institution-ID", paymentHubProperties.registeringInstitutionId());
+    httpHeaders.add("Purpose", "USCT Payment");
+    httpHeaders.add("X-CorrelationID", requestID);
+    httpHeaders.add("Platform-TenantId", paymentHubProperties.tenant());
+    httpHeaders.add("X-Program-ID", paymentHubProperties.programId());
+    httpHeaders.add("X-Signature", signOrderPaymentRequest(requestID, body));
+    httpHeaders.add("type", "raw");
+    httpHeaders.add("X-Road-Client", paymentHubBBInformationMediatorProperties.header());
+
+    var response =
+        new RestTemplate(requestFactory)
```

Comment:
Usually RestTemplate is created (or injected) in constructor and then shared and reused (it is somewhat heavyweight component). That is thread-safe if the configuration is not changed during the request.